lib/modules/datasource/cpan/index.ts

```diff
@@ -0,0 +1,147 @@
+import { logger } from '../../../logger';
+import { ExternalHostError } from '../../../types/errors/external-host-error';
+import { cache } from '../../../util/cache/package/decorator';
+import { getElapsedMinutes } from '../../../util/date';
+import { newlineRegex, regEx } from '../../../util/regex';
+import { copystr } from '../../../util/string';
+import { joinUrlParts } from '../../../util/url';
+import * as perlVersioning from '../../versioning/perl';
+import { Datasource } from '../datasource';
+import type { GetReleasesConfig, Release, ReleaseResult } from '../types';
+
+type Package = {
+  release: Release;
+  distribution: string;
+};
+
+let lastSync = new Date('2000-01-01');
+let packages: Record<string, Package> = Object.create(null); // Because we might need a "constructor" key
+let contentLength = 0;
+
+const pathExtensionPattern = regEx(/(\.tgz|\.tar\.gz)$/);
+const pathPattern = regEx(/^.+\/(.+)-(?:\d+(?:\.\d+)*)(?:\.tgz|\.tar\.gz)$/);
+
+// Note: use only for tests
+export function resetCache(): void {
+  lastSync = new Date('2000-01-01');
+  packages = Object.create(null);
+  contentLength = 0;
+}
+
+export class CpanDatasource extends Datasource {
+  static readonly id = 'cpan';
+
+  constructor() {
+    super(CpanDatasource.id);
+  }
+
+  override readonly defaultRegistryUrls = ['https://www.cpan.org'];
+
+  override readonly defaultVersioning = perlVersioning.id;
+
+  @cache({
+    namespace: `datasource-${CpanDatasource.id}`,
+    key: ({ packageName }: GetReleasesConfig) => `${packageName}`,
+  })
+  override async getReleases({
+    packageName,
+    registryUrl,
+  }: GetReleasesConfig): Promise<ReleaseResult | null> {
+    // istanbul ignore if
+    if (!registryUrl) {
+      return null;
+    }
+
+    await this.syncVersions(registryUrl);
```

Comment:
instead of using a global cache, return the cache from `updateCpanVersions` and add the cache decorator there too.

Reply:
Switching to use MetaCPAN api make this unnecessary.